src/Microsoft.VisualStudio.ProjectSystem.CSharp.VS.UnitTests/ProjectSystem/VS/Xproj/MigrateXprojProjectFactoryTests.cs

```diff
@@ -64,18 +99,143 @@ public void MigrateXprojProjectFactory_ValidPaths_CallMigrateCorrectly()
+            {
+                Assert.Equal("dotnet.exe", pr.FileName);
+                Assert.Equal("migrate -s -p \"C:\\Test\" -x \"C:\\Test\\XprojMigrationTests.xproj\"", pr.Arguments);
+            }, outputText: "Standard Output", errorText: "Standard Error");
+            var migrator = new MigrateXprojProjectFactory(procRunner);
+
+            var loggedMessages = new List<LogMessage>();
+            var logger = IVsUpgradeLoggerFactory.CreateLogger(loggedMessages);
+
+            Assert.True(migrator.MigrateProject(projectDirectory, xproj, "XprojMigrationTests", logger));
+            Assert.Equal(2, loggedMessages.Count);
+            Assert.Equal(new LogMessage
+            {
+                File = xproj,
+                Level = (uint)__VSUL_ERRORLEVEL.VSUL_INFORMATIONAL,
+                Message = "Standard Output",
+                Project = "XprojMigrationTests"
+            }, loggedMessages[0]);
+            Assert.Equal(new LogMessage
+            {
+                File = xproj,
+                Level = (uint)__VSUL_ERRORLEVEL.VSUL_WARNING,
+                Message = "Standard Error",
+                Project = "XprojMigrationTests"
+            }, loggedMessages[1]);
+        }
+
+        [Fact]
+        public void MigrateXprojProjectFactory_MigrateError_ReturnsFalse()
+        {
+            var projectDirectory = @"C:\Test";
+            var xproj = @"C:\Test\XprojMigrationTests.xproj";
+
+            // Runner returns valid response, standard exit code
+            var procRunner = ProcessRunnerFactory.ImplementRunner(pr =>
+            {
+                Assert.Equal("dotnet.exe", pr.FileName);
+                Assert.Equal("migrate -s -p \"C:\\Test\" -x \"C:\\Test\\XprojMigrationTests.xproj\"", pr.Arguments);
+            }, exitCode: VSConstants.E_FAIL);
+            var migrator = new MigrateXprojProjectFactory(procRunner);
+
+            var loggedMessages = new List<LogMessage>();
+            var logger = IVsUpgradeLoggerFactory.CreateLogger(loggedMessages);
+            Assert.False(migrator.MigrateProject(projectDirectory, xproj, "XprojMigrationTests", logger));
+            Assert.Equal(1, loggedMessages.Count);
+            Assert.Equal(new LogMessage
+            {
+                Level = (uint)__VSUL_ERRORLEVEL.VSUL_ERROR,
+                Project = "XprojMigrationTests",
+                File = xproj,
+                Message = $"Failed to migrate XProj project XprojMigrationTests. 'dotnet migrate -s -p \"{projectDirectory}\" -x \"{xproj}\"' exited with error code {VSConstants.E_FAIL}."
+            }, loggedMessages[0]);
+        }
+
+        [Fact]
+        public void MigrateXprojProjectFactory_ValidCsproj_FindsCsproj()
+        {
+            var tuple = CreateTempProjectLocation();
+            var xproj = tuple.Item2;
+            var projectDir = tuple.Item4;
+            var csproj = Path.Combine(projectDir, "XprojMigrationTests.csproj");
+
+            File.Create(csproj).Dispose();
```

Comment:
Until we have the actual migration report, which will tell us where the generated csproj is located and what it is named, we'll just search the folder for one. This is a test of that logic.